### PR TITLE
Make changes to reflect Metered Billing

### PR DESCRIPTION
**Pull request by panzors**
## Description
Adding metered capabilities for app store subscriptions end point

## Release Notes
Update to the latest payload schema

## Types of Changes

- [x] New feature (non-breaking change that adds functionality)

### CODE REVIEW

File: xero-app-store.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  version: "2.20.1"
+  version: "2.21.0"
```
Comment (panzors) [inline xero-app-store.yaml:3]: Bump Minor version as new fields are being added

File: xero-app-store.yaml [hunk 2]
```diff
@@ -262,15 +283,22 @@ components:
         name:
           description: The name of the product
           type: string
+        seatUnit:
```
Comment (panzors) [inline xero-app-store.yaml:286]: alphabetical ordering

File: xero-app-store.yaml [hunk 3]
```diff
@@ -229,6 +246,10 @@ components:
         product:
           description: The product subscribed to.
           $ref: "#/components/schemas/Product"
+        quantity:
+          description: The quantity of the item. For a fixed product, it is 1. For a per-seat product, it is a positive integer. For metered products, it is always null.
+          type: integer
+          nullable: true
```
Comment (RettBehrens) [inline xero-app-store.yaml:252]: Please remove `nullable: true`. It's used in the internal API spec but will break our generated SDKs. If a field is required it's called out in the schema, like so:

```
schemas:
    Subscription:
      type: object
      properties:
        currentPeriodEnd:
          description: |
            End of the current period that the subscription has been invoiced for.
          format: date-time
          type: string
        endDate:
          description: If the subscription has been canceled, this is the date when the subscription ends. If null, the subscription is active and has not been cancelled
          format: date-time
          type: string
        id:
          description: The unique identifier of the subscription
          format: uuid
          type: string
        organisationId:
          description: The Xero generated unique identifier for the organisation
          format: uuid
          type: string
        plans:
          description: List of plans for the subscription.
          type: array
          items:
            $ref: "#/components/schemas/Plan"
        startDate:
          description: Date when the subscription was first created.
          format: date-time
          type: string
        status:
          description: Status of the subscription. Available statuses are ACTIVE, CANCELED, and PAST_DUE.
          type: string
          enum:
          - ACTIVE
          - CANCELED
          - PAST_DUE
        testMode:
          description: Boolean used to indicate if the subscription is in test mode
          type: boolean
      required:
        - id
        - organisationId
        - status
        - startDate
        - currentPeriodEnd
        - plans
```

File: xero-app-store.yaml [hunk 4]
```diff
@@ -262,15 +283,22 @@ components:
         name:
           description: The name of the product
           type: string
+        seatUnit:
+          description: The unit of the per seat product. e.g. "user", "organisation", "SMS", etc
+          type: string
+          nullable: true
```
Comment (RettBehrens) [inline xero-app-store.yaml:289]: ditto

File: xero-app-store.yaml [hunk 5]
```diff
@@ -262,15 +283,22 @@ components:
...
           - PER_SEAT
-        seatUnit:
-          description: The unit of the per seat product. e.g. "user", "organisation", "SMS", etc
-          type: string
\ No newline at end of file
+          - METERED
+        usageUnit:
+          description: The unit of the usage product. e.g. "user", "minutes", "SMS", etc
+          type: string
+          nullable: true
```
Comment (RettBehrens) [inline xero-app-store.yaml:304]: ditto

**Review by rdemarco-xero — COMMENTED** [review]
I agree with Rett's comment around removing "nullable: true". Otherwise, looks good!

**Review by RettBehrens — APPROVED** [review]

**Pull request merged**